services/frontend-service/src/ui/components/SideBar/SideBar.test.tsx

```diff
@@ -565,4 +566,149 @@ describe('Action details', () => {
+            };
+            return (
+                <MemoryRouter>
+                    <SideBar {...defaultProps} {...overrides} />
+                </MemoryRouter>
+            );
+        };
+        const getWrapper = (overrides?: {}) => render(getNode(overrides));
+
+        describe.each(data)('', (testcase) => {
+            it(testcase.name, () => {
+                updateActions(testcase.actions);
+                const { container } = getWrapper({});
+                expect(container.getElementsByClassName('mdc-drawer-sidebar-header-title')[0].textContent).toBe(
+                    testcase.expectedTitle
+                );
+            });
+        });
+    });
+    describe('Sidebar shows updates number of planned actions', () => {
+        interface dataT {
+            name: string;
+            actions: BatchAction[];
+            expectedTitle: string;
+        }
+
+        const data: dataT[] = [
+            {
+                name: 'add 2 actions',
+                actions: [
+                    { action: { $case: 'undeploy', undeploy: { application: 'nmww' } } },
+                    { action: { $case: 'prepareUndeploy', prepareUndeploy: { application: 'nmww' } } },
+                ],
+                expectedTitle: 'Planned Actions (3)',
+            },
+            {
+                name: 'Add another action',
+                actions: [
+                    {
+                        action: {
+                            $case: 'deploy',
+                            deploy: {
+                                environment: 'foo',
+                                application: 'bread',
+                                version: 1337,
+                                ignoreAllLocks: false,
+                                lockBehavior: LockBehavior.Ignore,
+                            },
+                        },
+                    },
+                ],
+                expectedTitle: 'Planned Actions (4)',
+            },
+            {
+                name: 'Add 2 more actions actions',
+                actions: [
+                    { action: { $case: 'undeploy', undeploy: { application: 'test2' } } },
+                    { action: { $case: 'prepareUndeploy', prepareUndeploy: { application: 'test2' } } },
+                ],
+                expectedTitle: 'Planned Actions (6)',
+            },
+        ];
+
+        const getNode = (overrides?: {}): JSX.Element | any => {
+            // given
+            const defaultProps: any = {
+                children: null,
+            };
+            return (
+                <MemoryRouter>
+                    <SideBar {...defaultProps} {...overrides} />
+                </MemoryRouter>
+            );
+        };
+        const getWrapper = (overrides?: {}) => render(getNode(overrides));
+        it('Create an action initially', () => {
+            updateActions([{ action: { $case: 'undeploy', undeploy: { application: 'test' } } }]);
```

Comment:
for example, no need to create all the complicated data for the deploy and such. 

use any only in tests will make testing much easier. 

the `useNumberOfActions` hook only checks the length of the array so it doesn't matter what you put inside.

```suggestion
            updateActions(['foo', 'bar', 'baz'] as any); 
```

Reply:
Alright. I will take this into consideration in the future.